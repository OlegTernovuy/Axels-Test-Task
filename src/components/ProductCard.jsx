import { Card, CardActionArea, CardContent, CardMedia } from "@mui/material";
import React from "react";

const ProductCard = () => {
  return (
    <Card sx={{ maxWidth: 345 }}>
      <CardActionArea>
        <CardMedia component="img" height="140" image="" alt="product title" />
        <CardContent>
          <h2>Lorem</h2>
          <p>
            Lorem ipsum dolor sit amet consectetur adipisicing elit. Totam,
            illo.
          </p>
        </CardContent>
      </CardActionArea>
    </Card>
  );
};

export default ProductCard;
