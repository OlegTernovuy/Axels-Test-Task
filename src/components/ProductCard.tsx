import { useNavigate } from 'react-router-dom';
import {
  CardActionArea,
  CardContent,
  Typography,
} from '@mui/material';

import { StyledCard, StyledCardMedia } from '../styled/StyledProductCard';

import { IProduct } from '../types';

const ProductCard = (props: IProduct) => {
  const navigate = useNavigate();

  const { title, desc, id, img } = props;

  return (
    <StyledCard>
      <CardActionArea onClick={() => navigate(`/products/${id}`)}>
        <StyledCardMedia
          height='140'
          src={img}
          alt='product title'
        />
        <CardContent>
          <Typography variant='h5' component='h2'>
            {title}
          </Typography>
          <Typography variant='body2'>{desc}</Typography>
        </CardContent>
      </CardActionArea>
    </StyledCard>
  );
};

export default ProductCard;
