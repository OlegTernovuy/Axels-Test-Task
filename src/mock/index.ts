import { IComment } from '../types';

export const mockedComments: IComment[] = [
    {
        commentId: 1,
        commentText: 'Some comment',
        rating: 4,
    },
    {
        commentId: 2,
        commentText: 'Some another comment',
        rating: 5,
    },
    {
        commentId: 3,
        commentText: 'Good comment',
        rating: 5,
    },
];

export const mockedProduct = {
    id: 1,
    title: 'Prod 1',
    img: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQuMOUlcm_1UhO81Hul6fSb3rJgjrLeh1GcGQ&s',
    desc: 'Desk about prod 1',
    comments: [
        {
            commentId: 1,
            commentText: 'Some comment',
            rating: 4,
        },
        {
            commentId: 2,
            commentText: 'Some another comment',
            rating: 5,
        },
        {
            commentId: 3,
            commentText: 'Good comment',
            rating: 5,
        },
    ],
};

export const mockedProducts = [
    {
         "id": 1,
         "title": "Prod 1",
         "img": "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQuMOUlcm_1UhO81Hul6fSb3rJgjrLeh1GcGQ&s",
         "desc": "Desk about prod 1"
       },
       {
         "id": 2,
         "title": "Prod 2",
         "img": "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQuMOUlcm_1UhO81Hul6fSb3rJgjrLeh1GcGQ&s",
         "desc": "Desk about prod 2"
       },
       {
         "id": 3,
         "title": "Prod 3",
         "img": "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQuMOUlcm_1UhO81Hul6fSb3rJgjrLeh1GcGQ&s",
         "desc": "Desk about prod 3"
       },
       {
         "id": 4,
         "title": "Prod 4",
         "img": "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQuMOUlcm_1UhO81Hul6fSb3rJgjrLeh1GcGQ&s",
         "desc": "Desk about prod 4"
       },
       {
         "id": 5,
         "title": "Prod 5",
         "img": "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQuMOUlcm_1UhO81Hul6fSb3rJgjrLeh1GcGQ&s",
         "desc": "Desk about prod 5"
       }
      ]
