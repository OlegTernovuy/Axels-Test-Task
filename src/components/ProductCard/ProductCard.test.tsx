import { fireEvent, render, screen } from '@testing-library/react';

import ProductCard from './ProductCard';

import { mockedProduct } from '../../mock';

const mockedUsedNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: () => mockedUsedNavigate,
}));

describe('ProductCard component', () => {
    test('Navigate to product modal on click', () => {
        render(<ProductCard product={mockedProduct} />);
        fireEvent.click(screen.getByRole('button'));
        expect(mockedUsedNavigate).toHaveBeenCalledWith(
            `/products/${mockedProduct.id}`
        );
    });

    it('ProductCard matches Snapshot', () => {
        const view = render(<ProductCard product={mockedProduct} />);
        expect(view).toMatchSnapshot();
    });
});
