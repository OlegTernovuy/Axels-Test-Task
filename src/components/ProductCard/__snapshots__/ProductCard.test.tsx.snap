// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product card snapshot component Matches Snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root sc-aYaIB gErSSs css-bhp9pd-MuiPaper-root-MuiCard-root"
      >
        <button
          class="MuiButtonBase-root MuiCardActionArea-root css-1oitmrd-MuiButtonBase-root-MuiCardActionArea-root"
          tabindex="0"
          type="button"
        >
          <img
            alt="product title"
            class="sc-gEvDqW fUaAbP"
            height="140"
            src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQuMOUlcm_1UhO81Hul6fSb3rJgjrLeh1GcGQ&s"
          />
          <div
            class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
          >
            <h2
              class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
            >
              Prod 1
            </h2>
            <p
              class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
            >
              Desk about prod 1
            </p>
          </div>
          <span
            class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
          />
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root sc-aYaIB gErSSs css-bhp9pd-MuiPaper-root-MuiCard-root"
    >
      <button
        class="MuiButtonBase-root MuiCardActionArea-root css-1oitmrd-MuiButtonBase-root-MuiCardActionArea-root"
        tabindex="0"
        type="button"
      >
        <img
          alt="product title"
          class="sc-gEvDqW fUaAbP"
          height="140"
          src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQuMOUlcm_1UhO81Hul6fSb3rJgjrLeh1GcGQ&s"
        />
        <div
          class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <h2
            class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
          >
            Prod 1
          </h2>
          <p
            class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
          >
            Desk about prod 1
          </p>
        </div>
        <span
          class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
        />
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ProductCard component ProductCard matches Snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root sc-aYaIB gErSSs css-bhp9pd-MuiPaper-root-MuiCard-root"
      >
        <button
          class="MuiButtonBase-root MuiCardActionArea-root css-1oitmrd-MuiButtonBase-root-MuiCardActionArea-root"
          tabindex="0"
          type="button"
        >
          <img
            alt="product title"
            class="sc-gEvDqW fUaAbP"
            height="140"
            src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQuMOUlcm_1UhO81Hul6fSb3rJgjrLeh1GcGQ&s"
          />
          <div
            class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
          >
            <h2
              class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
            >
              Prod 1
            </h2>
            <p
              class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
            >
              Desk about prod 1
            </p>
          </div>
          <span
            class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
          />
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root sc-aYaIB gErSSs css-bhp9pd-MuiPaper-root-MuiCard-root"
    >
      <button
        class="MuiButtonBase-root MuiCardActionArea-root css-1oitmrd-MuiButtonBase-root-MuiCardActionArea-root"
        tabindex="0"
        type="button"
      >
        <img
          alt="product title"
          class="sc-gEvDqW fUaAbP"
          height="140"
          src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQuMOUlcm_1UhO81Hul6fSb3rJgjrLeh1GcGQ&s"
        />
        <div
          class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <h2
            class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
          >
            Prod 1
          </h2>
          <p
            class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
          >
            Desk about prod 1
          </p>
        </div>
        <span
          class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
        />
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
